### DISCUSSION TITLE
Sidef is Ruby?

### DISCUSSION THREAD

**Discussion by catb0t** [trizen/sidef#34] [General]
![noitsnot](https://cloud.githubusercontent.com/assets/3238748/19216717/f1016b14-8d95-11e6-8695-9346231e17b0.png)

[Here's](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language) how to fix it -- don't worry that Sidef might not meet their requirements for popularity -- amazingly active yet underappreciated [Factor](https://github.com/factor/factor), and Fantom, a long dead PHP-like for the JVM, both have entries in there despite small usage on GH.

The only possible obstacle I see is that a [grammar](https://github.com/github/linguist/blob/master/grammars.yml) in [TextMate grammar format](https://manual.macromates.com/en/language_grammars) must a) exist and b) **be licensed under one of the following**:

> - apache-2.0
> - bsd-2-clause
> - bsd-3-clause
> - permissive
> - mit
> - isc
> - mpl-2.0
> - unlicense
> - wtfpl
> - zlib

It omits it omits both versions of the Artistic license. 

Though, Perl and Perl 6 grammars are definitely supported on GH and almost definitely licensed under Artistic, so I'm not sure how that works...?

**Comment by trizen** [comment]
The Ruby shebang is used for syntax highlighting in Sidef scripts, since Ruby and Sidef share about the same syntax, but perhaps we can do better than this, as you described above.

The obstacle regarding the license of the TextMate grammar is not a big one, as only this grammar must be licensed under one of those licenses, not the whole Sidef project.

I'm not sure how this works either, as I only briefly looked over those links, but I think it should be possible to copy the TextMate grammar of Ruby and modify it for Sidef. If you plan to do this, I'm willing to get involved as well.

Thanks for the suggestions.

**Comment by catb0t** [comment]
I closed this to come back to it, and then forgot about it because preoccupied

I think if it's feasible to parse Sidef (reasonably) with a generic parser this should be not so hard.

**Comment by catb0t** [comment]
The thing about existing TMBundles is they don't contain the (old-style, readable) plist file used to generate them, only a lot of XML plists. Pretty hard to change the result of TextMate's compilation, but TextMate is mac-only  https://github.com/textmate/ruby.tmbundle
 
https://github.com/catb0t/sidef.tmbundle
